pkg/services/secrets/kvstore/remote_plugin.go

```diff
@@ -32,10 +41,13 @@ func (kv *secretsKVStorePlugin) Get(ctx context.Context, orgId int64, namespace
 		err = wrapUserFriendlySecretError(res.UserFriendlyError)
 	}
 
+	updateFatalFlag(ctx, *kv)
```

Comment:
Shouldn't this be only present on set? I don't see why retrieving a secret should result in fatal failure if compatibility is disabled.

Reply:
will huddle tomorrow to talk about it.